### PR TITLE
Add GitHub Action for Test & Release

### PR DESCRIPTION
**Pull request by alindeman**
Add a workflow for running tests.

Additionally add a workflow for building an RPM via `nfpm` and (if tagged) push an artifact as a release.

### CODE REVIEW

File: .github/workflows/package.yml [hunk 1]
```diff
@@ -0,0 +1,30 @@
+name: Package
+on:
+  push: {}
```
Comment (lstoll) [inline .github/workflows/package.yml:3]: i assume we want to run this all the time anyway to make sure packaging works, even if we dont actually "release" it?

**Review by lstoll — APPROVED** [review]
lgtm

File: .github/workflows/package.yml (hunk 1 above)
Comment (alindeman) [reply .github/workflows/package.yml:3]: Yah, that was my idea. If we only have it `on.push.tags` but it was broken in the intermediate time, we won't know until we push the tag that it's broken.

Do you think that's an OK way to approach it? Or is there a better way?

File: .github/workflows/package.yml [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+      - name: Install nfpm
+        run: go install github.com/goreleaser/nfpm/v2/cmd/nfpm@v2.6.0
+
+      - name: Build RPM
+        run: make rpm
+
+      - name: Release
+        uses: softprops/action-gh-release@v1
+        if: startsWith(github.ref, 'refs/tags/')
```
Comment (karlhungus) [inline .github/workflows/package.yml:27]: would it make sense to check that `hack/package_version.sh != "dev"`

Comment (karlhungus) [reply .github/workflows/package.yml:27]: derp, never mind

Comment (alindeman) [reply .github/workflows/package.yml:27]: We're pretty limited to what we can do in an `if` here. But in general, a release is 1:1 with a tag, so I think it's best to check for a tag here.

**Comment by alindeman** [comment]
Going to land this. I'm still not 100% sure it will work end-to-end, but I'll follow up with another PR if needed.

**Pull request merged**